tabs/src/components/SendPayment.tsx

```diff
@@ -149,100 +118,108 @@ const SendPayment: React.FC<SendPopupProps> = ({
-      const processedInvoice = data.split('lightning:').pop() || '';
-      console.log('QRCode scanned', processedInvoice);
-      if (processedInvoice) {
-        try {
-          const decodedInvoice = decode(processedInvoice);
-          console.log('QRCode decoded', decodedInvoice);
+  const decodeAndSetInvoice = async (processedInvoice: string) => {
+    try {
+      const decodedInvoice = decode(processedInvoice);
+      const amountSection = decodedInvoice.sections.find(
+        (section: any) => section.name === 'amount',
+      ) as { name: string; value: string } | undefined;
 
-          const amountSection = decodedInvoice.sections.find(
-            (section: any) => section.name === 'amount',
-          ) as { name: string; value: string };
+      const amountValue = amountSection ? parseInt(amountSection.value) : null;
+      const invoiceAmountInSatoshis = amountValue ? amountValue / 1000 : undefined;
 
-          // Access the amount value safely
-          const amountValue = parseInt(amountSection?.value ?? null);
+      const memoSection = decodedInvoice.sections.find(
+        (section: any) => section.name === 'description',
+      ) as { name: string; value: string } | undefined;
+      const memoValue = memoSection ? String(memoSection.value) : undefined;
 
-          console.log('Invoice amount (milliSats): ', amountValue); // Outputs: '181537000'
+      setInvoiceAmount(invoiceAmountInSatoshis);
+      setInvoiceMemo(memoValue);
+      setInvoice(processedInvoice);
+      setIsAmountReadOnly(true); // Make input read-only if needed
+    } catch (err) {
+      console.error('Error decoding invoice:', err);
+      setInvoiceAmount(undefined);
+      setInvoiceMemo(undefined);
+    }
+  };
 
-          // Only set state if the component is still mounted
-          if (isMounted.current) {
-            setDecodedInvoice(decodedInvoice);
-            if (decodedInvoice && amountValue) {
-              setInvoiceAmount(amountValue / 1000);
-              setIsAmountReadOnly(true);
-            }
-          }
-        } catch (err) {
-          console.error('Error decoding invoice:', err);
-          if (isMounted.current) {
-            setDecodedInvoice(null);
-          }
-        }
-      }
+  const handleScan = debounce(async (detectedCodes: IDetectedBarcode[]) => {
+    if (!isMounted.current) return;
+
+    if (detectedCodes.length > 0) {
+      const data = detectedCodes[0].rawValue;
+      const processedInvoice = data.split('lightning:').pop() || '';
 
-      if (isMounted.current) {
-        setInvoice(processedInvoice || '');
+      if (processedInvoice) {
+        await decodeAndSetInvoice(processedInvoice);
         setIsScanning(false);
+        setScannerPaused(true);
       }
     }
-  };
+  }, 500);
 
   const handleError = (error: any) => {
     console.error('QR Scan Error:', error);
+    if (error.name === 'NotAllowedError') {
+      alert(
```

Comment:
Is this like a JavaScript alert? Prefer to render the issue to the pop-up rather than an `alert()` like you have done in your other components.

Reply:
![image](https://github.com/user-attachments/assets/80f724ca-779e-4d47-b7da-6e5fadc368da)